### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by jorgemmsilva** [iotaledger/evm-toolkit#15]
### Bug description

when wallet is connected, if you try to save a new custom network, it will fail with:
```
: Expected an array with at least one valid string HTTPS url 'rpcUrls', Received:
undefined
```

### Expected behavior

it should disconnect the wallet instead of erroring.

There could be a textbox to specify a wasp-api, if that is specified, the wallet should be prompted to switch to that new jsonrpc.

### Can the issue reliably be reproduced?

Yes

### Language

Rust

### Language Version

NA

### Version

NA

### Hardware specifications

- Operating system:  
- RAM:  
- Cores:  
- Device:

**Comment by begonaalvarezd** [comment]
Hi @jorgemmsilva, thanks for reporting this error.

I feel like the form doesnt have the right label, where it says "network url" is where the rpc url goes
![image](https://github.com/iotaledger/evm-toolkit/assets/3624944/e28d9645-d9e2-468b-9c5e-dcaec740871e)
we could rename the label so its more clear, the network can not switch if you dont provide an rpc, so instead of allowing you to switch, we just keep you in the same network you were previously connected and display the error you mentioned

wdyt? would renaming the label be enough or are you referring to something else?

**Comment by jorgemmsilva** [comment]
renaming that would be nice, I agree.

But the main issue stands, if you're connected to the testnet with a wallet and try to set a new custom network without a rpc url, it will error and not let you

**Comment by begonaalvarezd** [comment]
> renaming that would be nice, I agree.
> 
> But the main issue stands, if you're connected to the testnet with a wallet and try to set a new custom network without a rpc url, it will error and not let you

`rpcs` is a required param, the error we display is thrown by metamask, you cant connect to a network without specifying at least 1 rpc https://docs.metamask.io/wallet/reference/rpc-api/#parameters-1 thats why we dont disconnect from the current connection

**Comment by begonaalvarezd** [comment]
> 🤷 but why not just save the custom network anyway? as it stands the modal just keeps opened for no reason.

I dont think that saving something that is not complete or correct is right. In this case, because the form validation is not succesful it doesnt get saved, hence the popup doesnt close, it is not great UX closing an reseting form popups when the action didnt succeed because you will need to open it again and fill it again to complete it correctly

**Comment by jorgemmsilva** [comment]
It works perfectly if you're not connected, that field is not even mandatory...
